### PR TITLE
USWDS-Site: Add documentation updates for uswds compile 1.2.0

### LINKED ISSUE #2749: USWDS-Site: Add documentation updates for uswds-compile 1.2.0
## Summary
Update documentation to match the changes in uswds-compile release 1.2.0. 

### Tasks
- [x] Update references to gulp 4
- [x] Add documentation for new `settings.compile.sassSourcemaps` setting
- [x] Add changelogs where needed

### PR DESCRIPTION
**Pull request by amyleadem**
# Summary
Updated documentation to match the changes in uswds-compile release 1.2.0. 

- Updated gulp 4 references to gulp 5
- Added documentation for new `settings.compile.sassSourcemaps` setting and  `settings.compile.browserslist` 
- Added changelogs where needed

> [!Important]
> We should update the changelog dates before merge.

## Related issue

Closes #2749 

## Preview link
[Getting started for developers: Phase 2](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-compile-1.2.0/documentation/getting-started/developers/phase-two-compile/)

## Testing and review
1. Confirm all doc changes related to uswds compile 1.2.0 have been accounted for
    1. All references to "Gulp 4" have been updated
    2. New compile settings are added to the path settings table
1. Confirm the changes in this PR are accurate and make sense
1. Confirm appropriate changelogs are present and accurate

### CODE REVIEW

File: pages/documentation/getting-started-developers/phase-two.md [hunk 1]
```diff
@@ -196,6 +196,8 @@ Setting | Default values - Version 2.x | Default values - Version 3.0 | Descript
 `paths.dist.js` | `"./assets/uswds/js"` | `"./assets/uswds/js"` | Project destination for compiled JavaScript
 `paths.dist.css` | `"./assets/uswds/css"` | `"./assets/uswds/css"` | Project destination for compiled CSS
 `sprite.projectIconsOnly` | `false` | `false` | Include _only_ the icons in `paths.src.projectIcons` in the icon sprite
+`settings.compile.browserslist` | `["> 2%", "last 2 versions", "IE 11", "not dead"]` | `["> 2%", "last 2 versions", "IE 11", "not dead"]` | Define the [browserslist query](https://github.com/browserslist/browserslist?tab=readme-ov-file#queries) for CSS prefixes generated by [autoprefixer](https://github.com/postcss/autoprefixer).
```
Comment (amyleadem) [inline pages/documentation/getting-started-developers/phase-two.md:199]: > [!note] 
> This setting was included in the README but not on the site. I added it here to fill the gap, even though it isn't related to 1.2.0

File: pages/documentation/getting-started-developers/phase-two.md [hunk 2]
```diff
@@ -196,6 +196,8 @@ Setting | Default values - Version 2.x | Default values - Version 3.0 | Descript
...
+`settings.compile.sassSourcemaps` | `true` | `true` | Include sourcemap when compiling SASS to CSS.
```
Comment (amyleadem) [inline pages/documentation/getting-started-developers/phase-two.md:200]: **question:** Now that we have multiple non-path settings available, should we create a separate table for them? Or rewrite the table header and intro?

File: pages/documentation/getting-started-developers/phase-two.md [hunk 3]
```diff
@@ -36,7 +36,7 @@ U.S. Web Design System source code is written in Sass, a powerful stylesheet lan
 
 We recommend using `uswds-compile`, a USWDS tool [hosted on GitHub](https://github.com/uswds/uswds-compile), as a quick way to get up and running with Sass compilation for USWDS.
 
-This tool provides simple Gulp 4 functions that copy USWDS static assets into your project and transform USWDS Sass into browser-readable CSS. Additionally, `uswds-compile` applies Autoprefixer to all compiled code to make sure your CSS is browser-compatible.
+This tool provides simple Gulp 5 functions that copy USWDS static assets into your project and transform USWDS Sass into browser-readable CSS. Additionally, `uswds-compile` applies Autoprefixer to all compiled code to make sure your CSS is browser-compatible.
```
Comment (amyleadem) [inline pages/documentation/getting-started-developers/phase-two.md:39]: **question**: Curious if you all think the specificity of the major version number is needed here. Can we just replace it with "Gulp"?

```suggestion
This tool provides simple Gulp functions that copy USWDS static assets into your project and transform USWDS Sass into browser-readable CSS. Additionally, `uswds-compile` applies Autoprefixer to all compiled code to make sure your CSS is browser-compatible.
```

File: pages/documentation/getting-started-developers/phase-two.md (hunk 2 above)
Comment (mejiaj) [reply pages/documentation/getting-started-developers/phase-two.md:200]: I'd keep it together for now. There aren't enough issues to warrant a completely separate table _yet_.

File: pages/documentation/getting-started-developers/phase-two.md (hunk 3 above)
Comment (mejiaj) [reply pages/documentation/getting-started-developers/phase-two.md:39]: I like this idea, less of a chance it gets missed in the future.

**Review by mejiaj — APPROVED** [review]
Thanks for updating this!

**Comment by mejiaj** [comment]
CI test failing due to snyk errors. Fixed by #2751.

File: _data/changelogs/docs-getting-started-devs-phase-2.yml [hunk 4]
```diff
@@ -2,6 +2,16 @@ title: "Phase 2: Compile"
 type: documentation
 changelogURL:
 items:
+  - date: NNNN-NN-NN
+    summary: Added documentation for `settings.compile.browserslist` and `settings.compile.sassSourcemaps`.
+    affectsGuidance: true
+    githubPr: 2750
+    githubRepo: uswds-site
+  - date: NNNN-NN-NN
+    summary: Replaced "Gulp 4" references with "Gulp 5".
```
Comment (mahoneycm) [inline _data/changelogs/docs-getting-started-devs-phase-2.yml:11]: **suggestion:** If we remove the gulp version from the documentation we should update this summary as well

File: pages/documentation/getting-started-developers/phase-two.md (hunk 1 above)
Comment (mahoneycm) [reply pages/documentation/getting-started-developers/phase-two.md:199]: **Praise:** Great catch! Thanks for adding this

**Review by mahoneycm — COMMENTED** [review]
Looking good! Once you remove the gulp version number this will be good to go in my eyes 👍

File: pages/documentation/getting-started-developers/phase-two.md (hunk 3 above)
Comment (amyleadem) [reply pages/documentation/getting-started-developers/phase-two.md]: Replaced "Gulp 5" with "Gulp" in [4426600](https://github.com/uswds/uswds-site/pull/2750/commits/44266000175d05ec29501f17b4187efab1b15f53)

File: _data/changelogs/docs-getting-started-devs-phase-2.yml (hunk 4 above)
Comment (amyleadem) [reply _data/changelogs/docs-getting-started-devs-phase-2.yml]: Good flag! Updated the summary here to "Removed the version number from "Gulp 4" references." in [4426600](https://github.com/uswds/uswds-site/pull/2750/commits/44266000175d05ec29501f17b4187efab1b15f53)

**Review by mahoneycm — APPROVED** [review]

**Review by mejiaj — APPROVED** [review]
Thanks @amyleadem! Didn't find any visual regressions in preview.

File: _data/changelogs/docs-getting-started-devs-phase-2.yml [hunk 5]
```diff
@@ -2,6 +2,16 @@ title: "Phase 2: Compile"
 type: documentation
 changelogURL:
 items:
+  - date: NNNN-NN-NN
+    summary: Added documentation for `settings.compile.browserslist` and `settings.compile.sassSourcemaps`.
+    affectsGuidance: true
+    githubPr: 2750
+    githubRepo: uswds-site
+  - date: NNNN-NN-NN
```
Comment (thisisdano) [inline _data/changelogs/docs-getting-started-devs-phase-2.yml:10]:
```suggestion
  - date: 2024-09-04
```

**Review by thisisdano — COMMENTED** [review]
Updating the changelog date

File: _data/changelogs/docs-getting-started-devs-phase-2.yml [hunk 6]
```diff
@@ -2,6 +2,16 @@ title: "Phase 2: Compile"
 type: documentation
 changelogURL:
 items:
+  - date: NNNN-NN-NN
+    summary: Added documentation for `settings.compile.browserslist` and `settings.compile.sassSourcemaps`.
+    affectsGuidance: true
+    githubPr: 2750
+    githubRepo: uswds-site
+  - date: 2024-09-04
```
Comment (thisisdano) [inline _data/changelogs/docs-getting-started-devs-phase-2.yml:10]:
```suggestion
  - date: 2024-09-05
```

File: _data/changelogs/docs-getting-started-devs-phase-2.yml [hunk 7]
```diff
@@ -2,6 +2,16 @@ title: "Phase 2: Compile"
 type: documentation
 changelogURL:
 items:
+  - date: NNNN-NN-NN
```
Comment (thisisdano) [inline _data/changelogs/docs-getting-started-devs-phase-2.yml:5]:
```suggestion
  - date: 2024-09-05
```

**Review by thisisdano — APPROVED** [review]

**Pull request merged**